server/api/src/rsDataObjRead.cpp

```diff
@@ -114,10 +116,27 @@ rsDataObjRead(rsComm_t* rsComm,
             return 0;
         }
 
+        // Store into 'dst' (an integer of bitwidth B) the value of 'src' (bitwidth >= B))
+        //   or the maximum value that will fit into dst, whichever is less.
+
+        auto limit_to_dsttype = [](auto & dst, const auto& src) noexcept {
+            using srcI = std::remove_reference_t< decltype(src) >;
+            using dstI = std::remove_reference_t< decltype(dst) >;
+            static_assert( std::is_signed_v<srcI> == std::is_signed_v<dstI>
```

Comment:
Why do we need a static_assert here? It the data types ever result in a failure, we'd have to reevaluate this.

Reply:
I was just curious.

Consider using a better name than `t`. Perhaps, `new_buffer_size`, `corrected_length`, etc.

Reply:
Agreed, using `new_buffer_size` - and,this way is less confusing.